docs/dev/zksync-v2/aa.md

```diff
@@ -10,41 +10,103 @@ Accounts in zkSync 2.0 can initiate transactions, like an EOA, but can also have
+The account abstraction protocol on zkSync is in spirit very similar to [EIP4337](https://eips.ethereum.org/EIPS/eip-4337), though our protocol is still different for the sake of efficiency and better UX.
 
-## Fees
+### IAccount interface
 
-The system charges fees by doing an ERC20/ETH `transfer` from an account to the operator's account.
+Each account is recommended to implement the [IAccount](https://github.com/matter-labs/v2-testnet-contracts/blob/main/zksync/system-contracts/interfaces/IAccount.sol) interface. It contains the following five methods:
 
-In the EIP4337 you can see three types of gas limits: `verificationGas`, `executionGas`, `preVerificationGas`, that describe the gas limit for the different steps of the transaction's inclusion in a block. 
-Currently, zkSync supports only a single field, `ergsLimit`, that covers the fee for all three. When submitting a transaction make sure that `ergsLimit` is enough to cover verification, paying the fee (the ERC20 transfer mentioned above) and the actual execution itself.
+- `validateTransaction` is mandatory and will be used by the system to determine if the AA logic agrees to proceed with the transaction. In case the transaction is not accepted (e.g. the signature is wrong) the method should revert. In case the call to this method succeedes, the implemented account logic is considered to accept the transaction, and the system will proceed with the transaction flow.
+- `executeTransaction` is mandatory and will be called by the system after the fee is charged from the user. This function should perform the execution of the transaction.
+- `payForTransaction` is optional and will be called by the system if the transaction has no paymaster, i.e. the account is willing to pay for the transaction. This method should be used to pay for the fees by the account. Note, that if your account will never pay any fees and will always rely on the [paymaster](#paymasters) feature, you don't have to implement this method. This method must send at least `tx.gasprice * tx.ergsLimit` ETH to the [bootloader](./system-contracts.md#bootloader) address.
+- `prePaymaster` is optional and will be called by the system if the transaction has a paymaster, i.e. there is someone else who will pay the transaction fees for the user. This method should be used to prepare for the interaction with the paymaster. One of the notable [examples](#approval-based-paymaster-flow) where it can be helpful is to approve the ERC-20 tokens for the paymaster.
+- `executeTransactionFromOutside`, technically, is not mandatory, but it is _highly encouraged_, since there needs to be some way, in case of priority mode (e.g. when the operator becomes malicious), to be able to start transactions from your account from ``outside'' (basically this is the fallback to the standard Ethereum approach, where an EOA starts transaction from your smart contract).
```

Comment:
"when operator becomes malicious" is a bit depressing, and I'm not aware of such plans :)
Maybe "if the network can no longer process L2 transactions" or something similarly neutral?

Reply:
What about "e.g. if the operator is unresponsive"?

Reply:
lgtm